sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/DataSourceV2Strategy.scala

```diff
@@ -221,6 +221,12 @@ class DataSourceV2Strategy(session: SparkSession) extends Strategy with Predicat
                   throw new AnalysisException(s"Exec update failed:" +
                       s" cannot translate expression to source filter: $f"))
               }).toArray
+
+          if (!table.asDeletable.canDeleteWhere(filters)) {
+            throw new AnalysisException(
```

Comment:
is this exception handled later? the rewrite part for row deletion is a TBD?

Reply:
The rewrite would happen earlier. This just throws a good error message if `deleteWhere` will fail.

Reply:
The rewrite part is yet to be done. This PR just adds a way to have more info at planning time.

Reply:
I see, thanks. So this method will be called in an earlier place and before rewrite once the rewrite part is ready, is that right?

Reply:
It is going to be called at planning time to check if we should apply the rewrite or just pass filters down.